kevm-pyk/src/kevm_pyk/kproj/evm-semantics/lemmas/bytes-simplification.k

```diff
@@ -59,11 +59,17 @@ module BYTES-SIMPLIFICATION [symbolic]
       requires lengthBytes(B) <=Int 32
        andBool #rangeUInt(256, X) [simplification, concrete(B), comm]
 
-    rule [buf-asWord-invert-lr]:
+    rule [buf-asWord-invert-lr-len-leq]:
       #buf (W:Int , #asWord(B:Bytes)) => #buf(W -Int lengthBytes(B), 0) +Bytes B
       requires lengthBytes(B) <=Int W andBool W <=Int 32
       [simplification]
 
+    rule [buf-asWord-invert-lr-len-gt]:
+      #buf (W:Int , #asWord(B:Bytes)) => #range(B, lengthBytes(B) -Int W, W)
+      requires W <Int lengthBytes(B) andBool lengthBytes(B) <=Int 32
+       andBool #asWord(B) <Int 2 ^Int (8 *Int W)
+      [simplification]
```

Comment:
```suggestion
      [simplification, concrete(W)]
```
Since `W` it's used in a non-linear constraint in the previous line. This maybe limits expressivity a bit, but I think most of the time our `#buf`s have concrete width anyway, so I wouldn't expect to make a big difference.

Reply:
I would like to keep `W` concrete for now and remove `concrete` only if we find a use case in the real-world code. There is a number of other simplifications that follow this pattern.